core/services/llo/onchain_channel_definition_cache.go

```diff
@@ -0,0 +1,252 @@
+		}
+		c.wg.Add(1)
+		go c.poll()
+		return nil
+	})
+}
+
+// TODO: make this configurable?
+const pollInterval = 1 * time.Second
+
+func (c *channelDefinitionCache) poll() {
+	defer c.wg.Done()
+
+	pollT := time.NewTicker(utils.WithJitter(pollInterval))
+
+	for {
+		select {
+		case <-c.chStop:
+			return
+		case <-pollT.C:
+			if n, err := c.fetchFromChain(); err != nil {
+				// TODO: retry with backoff?
+				// https://smartcontract-it.atlassian.net/browse/MERC-3653
+				c.lggr.Errorw("Failed to fetch channel definitions from chain", "err", err)
+				continue
+			} else {
+				if n > 0 {
+					c.lggr.Infow("Updated channel definitions", "nLogs", n, "definitionsBlockNum", c.definitionsBlockNum)
+				} else {
+					c.lggr.Debugw("No new channel definitions", "nLogs", 0, "definitionsBlockNum", c.definitionsBlockNum)
+				}
+			}
+		}
+	}
+}
+
+func (c *channelDefinitionCache) fetchFromChain() (nLogs int, err error) {
+	// TODO: Pass context
+	// https://smartcontract-it.atlassian.net/browse/MERC-3653
+	latest, err := c.lp.LatestBlock()
+	if errors.Is(err, sql.ErrNoRows) {
+		c.lggr.Debug("Logpoller has no logs yet, skipping poll")
+		return 0, nil
+	} else if err != nil {
+		return 0, err
+	}
+	toBlock := latest.BlockNumber
+
+	fromBlock := c.definitionsBlockNum
+
+	if toBlock <= fromBlock {
+		return 0, nil
+	}
+
+	ctx, cancel := services.StopChan(c.chStop).NewCtx()
+	defer cancel()
+	// NOTE: We assume that log poller returns logs in ascending order chronologically
+	logs, err := c.lp.LogsWithSigs(fromBlock, toBlock, allTopics, c.addr, pg.WithParentCtx(ctx))
+	if err != nil {
+		// TODO: retry?
+		// https://smartcontract-it.atlassian.net/browse/MERC-3653
+		return 0, err
+	}
+	for _, log := range logs {
+		if err = c.applyLog(log); err != nil {
+			return 0, err
+		}
+	}
+
+	// Use context.Background() here because we want to try to save even if we
+	// are closing
+	if err = c.orm.StoreChannelDefinitions(context.Background(), c.addr, c.Definitions(), toBlock); err != nil {
+		return 0, err
+	}
+
+	c.definitionsBlockNum = toBlock
+
+	return len(logs), nil
+}
+
+func (c *channelDefinitionCache) applyLog(log logpoller.Log) error {
+	switch log.EventSig {
+	case topicNewChannelDefinition:
+		unpacked := new(channel_config_store.ChannelConfigStoreNewChannelDefinition)
+
+		err := channelConfigStoreABI.UnpackIntoInterface(unpacked, "NewChannelDefinition", log.Data)
+		if err != nil {
+			return fmt.Errorf("failed to unpack log data: %w", err)
+		}
+
+		c.applyNewChannelDefinition(unpacked)
+	case topicChannelDefinitionRemoved:
+		unpacked := new(channel_config_store.ChannelConfigStoreChannelDefinitionRemoved)
+
+		err := channelConfigStoreABI.UnpackIntoInterface(unpacked, "ChannelDefinitionRemoved", log.Data)
+		if err != nil {
+			return fmt.Errorf("failed to unpack log data: %w", err)
+		}
+
+		c.applyChannelDefinitionRemoved(unpacked)
+	default:
+		// don't return error here, we want to ignore unrecognized logs and
```

Comment:
If you wanna disable a channel you might as well just remove it. What would be the benefit of disabling instead of removing?

Doesn't really need the start block/time, it will start working automatically once `> 2f` nodes get the updated channel definitions.